pkg/driver/controller.go

```diff
@@ -452,7 +455,11 @@ func (d *Driver) doCreateVolume( //revive:disable-line:unused-receiver
 		// snapshot deletion strategy this will also prevent creation of
 		// identically named clone volumes in the future due to snapshot name
 		// collisions (see note below on cleanup).
-		snapName := "snapshot-" + req.Name
+
+		// Appending current time-unit and randome number to the snapshot name to make it unique.
+		// e.g snapshot-<volume-name>-<current-time> <31-bit non-negative-int> snapshot-volName-YYYYMMDDHHMMSS-1279846997  
+		formattedTime := time.Now().Format("20060102150405")
+		snapName := "snapshot-" + req.Name + "-" + formattedTime + "-" + strconv.Itoa(int(rand.Int31()))
```

Comment:
UUIDv7 is exactly doing this: https://uuid7.com/ and the uuid library you are using is already capable of generating them: https://github.com/google/uuid/blob/master/version7.go

In my POV this would make the code easier to understand and also follow a well known and documented standard.

Reply:
I see your point.